baitap-submit/ten_cua_ban/02-llm-api-params/src/fetch_article.py

```diff
@@ -0,0 +1,55 @@
+from bs4 import BeautifulSoup
+import urllib.request
+import re
+
+
+def fetch_article(url):
+    try:
+        # Send a GET request to the website
+        response = urllib.request.urlopen(url)
+        
+        # Check if the request was successful
+        if response.status == 200:
+            # Get the website content (HTML)
+            website_content = response.read().decode('utf-8')
+            
+            # Parse the HTML content using BeautifulSoup
+            soup = BeautifulSoup(website_content, 'html.parser')
+            
+            # Extract the article content
+            article = soup.find('div', id='main-detail')
+            if article:
+                print("Article content fetched successfully!")
+                return article.get_text(strip=True)  # Return the text content of the article
+            else:
+                print("Failed to find the article content.")
+                return None
+        else:
+            print(f"Failed to fetch the website. Status code: {response.status}")
+            return None
+
+    except urllib.error.URLError as e:
+        print(f"An error occurred: {e}")
+        return None
+
+def summarize_article(url):
+    """
+    Fetches the article content from the given URL and summarizes it.
+    """
+    print("\nFetching article content...")
+    article_content = fetch_article(url)
+    if article_content:
+        print("\nSummarizing article content...")
+        summary_prompt = f"Summarize the following article in its original language, maintaining the tone, context, and key points. Ensure the summary is concise but retains all critical information and does not change the meaning of the original text. Do not translate or alter the language of the article:\n\n{article_content}"
```

Comment:
{article_content} nên dc bỏ trong 
```
{context}
```. hoặc 
<input_text>
{context}
</input_text>

Trong python e có thể dùng multine string, ví dụ 
a = """Lorem ipsum dolor sit amet,
consectetur adipiscing elit,
sed do eiusmod tempor incididunt
ut labore et dolore magna aliqua."""


Này có thể sửa lại thành như vầy
summary_prompt = f"""Summarize the following article in its original language, maintaining the tone, context, and key points. Ensure the summary is concise but retains all critical information and does not change the meaning of the original text. Do not translate or alter the language of the article:\n\n
```
{article_content}
```
"""